src/Rhone/JS/Sink.idr

```diff
@@ -235,3 +235,27 @@ namespace LocalStorage
   export %inline
   setItemAt : LiftJSIO m => (key : String) -> MSF m String ()
   setItemAt = firstArg setItem
+
+--------------------------------------------------------------------------------
+--          Focus
+--------------------------------------------------------------------------------
+
+public export
+interface SafeCast t => Focus t where
+  focus' : t -> JSIO ()
+
+public export
+JSType t => Elem HTMLOrSVGElement (Types t) => SafeCast t => Focus t where
+  focus' = HTMLOrSVGElement.focus'
```

Comment:
Do we need an interface for this? The problem with this implementation seems to be that the `JSType` and `Elem` constraints are not erased, which - in case of `JSType` - leads to all the types collected in the `parents` and `mixins` fields lying around at runtime. I tried to annotate these constraints with a zero quantity, but the parser doesn't seem to accept these.

However, the following does seem to work:

```idris
public export
interface SafeCast t => Focus t where
  constructor MkFocus
  focus' : t -> JSIO ()

public export %hint
htmlFocusImpl :  (0 _ : JSType t)
              => (0 _ : Elem HTMLOrSVGElement (Types t))
              => SafeCast t
              => Focus t
htmlFocusImpl = MkFocus { focus' = HTMLOrSVGElement.focus' }
```

So, I suggest we either get rid of interface `Focus` entirely and use the necessary constraints in the `focus` sink directly, or we switch to a hand-written hint with the `JSType` and `Elem` constraints having zero quantity.

I quickly tested the second approach and whether Idris finds the hint and it seems to work.

Reply:
I think ditching the interface entirely probably makes sense to me. The interface had small readability benefits but nothing ground breaking. I pushed that change.